python_analytics/event_encoder.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import, unicode_literals
+
+from abc import ABCMeta, abstractmethod
+from weakref import WeakKeyDictionary
+
+from six import add_metaclass, text_type
+
+
+NoValue = object()
+
+
+@add_metaclass(ABCMeta)
+class _TrackedAttribute(object):
```

Comment:
I was playing with the idea of introducing `Int`, `String`, `Boolean` and the other types supported by Google Analytics. Do you think that would be better?

Instead of
```
class SomeEvent(object):
    foo = Parameter('ec', int)
```
you have
```
class Foo(object):
    foo = Int('ec')
```